app/src/main/java/com/firebase/samples/logindemo/MainActivity.java

```diff
@@ -220,15 +223,16 @@ public void onClick(View view) {
         mAuthProgressDialog.setCancelable(false);
         mAuthProgressDialog.show();
 
-        /* Check if the user is authenticated with Firebase already. If this is the case we can set the authenticated
-         * user and hide hide any login buttons */
-        mFirebaseRef.addAuthStateListener(new Firebase.AuthStateListener() {
+        firebaseAuthStateListener = new Firebase.AuthStateListener() {
             @Override
             public void onAuthStateChanged(AuthData authData) {
                 mAuthProgressDialog.hide();
                 setAuthenticatedUser(authData);
             }
-        });
+        };
+        /* Check if the user is authenticated with Firebase already. If this is the case we can set the authenticated
+         * user and hide hide any login buttons */
+        mFirebaseRef.addAuthStateListener(firebaseAuthStateListener);
```

Comment:
Another option would be attaching the listener only the first time the Activity is created (savedInstanceState == null). And never removing it. I think both solutions would work